sklearn/cluster/_affinity_propagation.py

```diff
@@ -178,6 +178,23 @@ def _affinity_propagation(
 # Public API
 
 
+@validate_params(
+    {
+        "S": ["array-like"],
+        "preference": [
+            "array-like",
+            Interval(Real, None, None, closed="neither"),
+            None,
+        ],
+        "convergence_iter": [Interval(Integral, 1, None, closed="left")],
+        "max_iter": [Interval(Integral, 1, None, closed="left")],
+        "damping": [Interval(Real, 0.5, 1.0, closed="both")],
+        "copy": ["boolean"],
+        "verbose": ["boolean"],
+        "return_n_iter": ["boolean"],
+        "random_state": ["random_state"],
+    }
+)
```

Comment:
I think that we only have to validate these two parameters which are not checked by the class `AffinityPropagation`

```suggestion
@validate_params({"S": ["array-like"], "return_n_iter": ["boolean"]})
```

Reply:
Done.